DataFormats/Calibration/include/DataFormatsCalibration/MeanVertexObject.h

```diff
@@ -46,42 +56,81 @@ class MeanVertexObject
   MeanVertexObject& operator=(MeanVertexObject& other) = default;
   MeanVertexObject& operator=(MeanVertexObject&& other) = default;
 
-  void setX(float val) { mPos[0] = val; }
-  void setY(float val) { mPos[1] = val; }
-  void setZ(float val) { mPos[2] = val; }
-  void setPos(std::array<float, 3> val)
+  void set(int icoord, float val)
   {
-    for (int i = 0; i < 3; i++) {
-      mPos[i] = val[i];
+    if (icoord == 0) {
+      setX(val);
+    } else if (icoord == 1) {
+      setY(val);
+    } else if (icoord == 2) {
+      setZ(val);
+    } else {
+      LOG(fatal) << "Coordinate out of bound to set vtx " << icoord << ", should be in [0, 2]";
```

Comment:
It's probably a good idea to avoid using Logger in an inlined function in an header. Why do you need to change the API?

Reply:
which API do you mean?

Reply:
Why setting things one by one rather than using the array?

Reply:
This sets only 1 coordinate.

Reply:
Yes, I get that, but why? Where is it actually used?

Reply:
See line 293 of Detectors/Calibration/src/MeanVertexCalibrator.cxx in this PR.
And anyway, I would not see any issue in having such method.